src/ui/public/indexed_array/indexed_array.js

```diff
@@ -33,7 +33,24 @@ export class IndexedArray {
     this._indexNames = _.union(
       this._setupIndex(config.group, inflectIndex, _.organizeBy),
       this._setupIndex(config.index, inflectIndex, _.indexBy),
-      this._setupIndex(config.order, inflectOrder, _.sortBy)
+      this._setupIndex(config.order, inflectOrder, (raw, pluckValue) => {
+        return raw.slice().sort((itemA, itemB) => {
```

Comment:
nit: we should be using the spread operator to copy Arrays per our styleguide [here](https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#use-the-spread-operator--for-copying-arrays)